responses/__init__.py

```diff
@@ -968,7 +968,7 @@ def add_callback(
         content_type: Optional[str] = "text/plain",
         match: "_MatcherIterable" = (),
     ) -> None:
```

Comment:
```suggestion
    ) -> BaseResponse:
```

The type annotations will need to be updated.